Dockerfile

```diff
@@ -1,18 +1,24 @@
-FROM alpine:edge
+FROM alpine:latest
+
 LABEL maintainer="dev@jpillora.com"
-# webproc release settings
-ENV WEBPROC_VERSION 0.2.2
-ENV WEBPROC_URL https://github.com/jpillora/webproc/releases/download/$WEBPROC_VERSION/webproc_linux_amd64.gz
+
+# expose ports
+EXPOSE 53
+EXPOSE 8080
+
 # fetch dnsmasq and webproc binary
-RUN apk update \
-	&& apk --no-cache add dnsmasq \
-	&& apk add --no-cache --virtual .build-deps curl \
-	&& curl -sL $WEBPROC_URL | gzip -d - > /usr/local/bin/webproc \
-	&& chmod +x /usr/local/bin/webproc \
-	&& apk del .build-deps
+RUN apk update && \
+	apk --no-cache add dnsmasq bash && \
+	apk add --no-cache --virtual .build-deps curl && \
+	curl -sL https://i.jpillora.com/webproc | bash && \
+	apk del .build-deps && \
+	mv webproc /usr/local/bin/ && \
+	\
+	mkdir -p /etc/default/ && \
+	echo -e "ENABLED=1\nIGNORE_RESOLVCONF=yes" > /etc/default/dnsmasq
+
 #configure dnsmasq
-RUN mkdir -p /etc/default/
-RUN echo -e "ENABLED=1\nIGNORE_RESOLVCONF=yes" > /etc/default/dnsmasq
```

Comment:
do we not need this file anymore?

Reply:
@macwinnie ?

Reply:
They moved these two `RUN` lines into the first `RUN` to keep them all in the same layer.

They droped the two `ENV` declarations at the start of the file to call some shell script file you're hosting at `https://i.jpillora.com/webproc`, piping that through bash to retrieve the latest version of webproc that way. While nice, it will remove immutability of the build as the version is no longer pinned.

That script does some of the logic here, unarchiving the file and outputting a binary with execution rights enabled via `chmod +x`, it just needs to move the binary.

The `ENTRYPOINT` change is just whitespace formatting and renaming the `--config` option to match the newer option name you must have modified while developing [webproc](https://github.com/jpillora/webproc).

Addition of `EXPOSE` only serves value of [communicating what ports the user should map to the image](https://docs.docker.com/engine/reference/builder/#expose), it doesn't really do any thing beyond that.